### PR TITLE
Howto loki

### LINKED ISSUE #12587: docs: Metrics article mentions missing grafana board and the document says nothing about Loki datasource for the new board
*Please describe the question or issue you're facing with "How to monitor metrics - Canonical LXD documentation".*

Article describes LXD, Prometheus and Grafana configuration to have Grafana dashboard to display LXD cluster parameters.
1. Screenshot mentions dashboard 15726 which is no longer available
2. Text mentions dashboard 19131, which has been changed to have both Prometheus and Loki input sources, but the article says nothing about Loki data source. 
Loki datasource has been introduced to the dashboard in September 2023 and it is a required parameter in Grafana. So at the moment the article cannot be used to configure Grafana.

---
*Reported from: https://documentation.ubuntu.com/lxd/en/latest/metrics/*

### PR DESCRIPTION
**Pull request by escabo**
Added Loki howto including instructions from this request: https://discourse.ubuntu.com/t/how-to-set-labels-for-loki/42317/10
Also fixes #12587

### CODE REVIEW

**Comment by tomponline** [comment]
@simondeziel @escabo is a member of the LXD team, but its showing him as not passing the CLA check, any ideas?

**Comment by tomponline** [comment]
Doc checks failing with:

```
 .tmp/doc/howto/logs_loki.md:60: MD012 Multiple consecutive blank lines
.tmp/doc/howto/logs_loki.md:61: MD012 Multiple consecutive blank lines
```

**Review by tomponline — COMMENTED** [review]
Because the docs go into the snap, any practical reduction in image size would be appreciated.

**Comment by tomponline** [comment]
Please can you remove the merge commit `749c5a3db564cbf767f8a7025447cb32a7fbd07e` from this PR?

**Comment by simondeziel** [comment]
> Because the docs go into the snap, any practical reduction in image size would be appreciated.

I vaguely remember that those PNGs were minified prior to adding them. Could you do the same with those new/updated ones, please?

File: doc/howto/grafana.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+   1. From the Basic (quick setup) panel, chose {guilabel}`Data Sources`.
+
+      ![Chose data source in Grafana](/images/grafana_welcome.png)
+
+   1. Select {guilabel}`Prometheus`.
+
+      ![Select Prometheus as a data source](/images/grafana_select_prometheus.png)
+
+   1. In the {guilabel}`URL` field, enter the address of your Prometheus install `http://localhost:9090/`.
```
Comment (ru-fu) [inline doc/howto/grafana.md:32]:
```suggestion
   1. In the {guilabel}`URL` field, enter the address of your Prometheus installation (`http://localhost:9090/`).
```

Comment (ru-fu) [reply doc/howto/grafana.md:32]: With or without the slash at the end? The screenshot has it without.

File: doc/howto/grafana.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+   1. Keep the default configuration for the other fields and click {guilabel}`Save & test`.
+
+1. Configure Loki as another data source:
+
+   1. Select {guilabel}`Loki`.
+
+      ![Select Loli as another data source](/images/grafana_select_loki.png)
+
+   1. In the {guilabel}`URL` field, enter the address of your Loki install `http://localhost:3100/`.
```
Comment (ru-fu) [inline doc/howto/grafana.md:44]:
```suggestion
   1. In the {guilabel}`URL` field, enter the address of your Loki installation (`http://localhost:3100/`).
```

Comment (ru-fu) [reply doc/howto/grafana.md:44]: Same question about the slash.

File: doc/howto/grafana.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+      ![Enter Loki URL](/images/grafana_configure_loki.png)
+
+   1. Keep the default configuration for the other fields and click {guilabel}`Save & test`.
+
+1. Import the LXD dashboard:
+
+   1. Go back to the Basic (quick setup) panel and now chose {guilabel}`Dashboards` > {guilabel}`Import a dashboard`.
```
Comment (ru-fu) [inline doc/howto/grafana.md:52]:
```suggestion
   1. Go back to the Basic (quick setup) panel and now choose {guilabel}`Dashboards` > {guilabel}`Import a dashboard`.
```

File: doc/howto/logs_loki.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+2024-02-14T21:31:20Z {app="lxd", instance="node3", type="logging"} level="info" Expiring log files
+2024-02-14T21:31:20Z {app="lxd", instance="node3", type="logging"} level="info" Pruning resolved warnings
+2024-02-14T21:31:20Z {app="lxd", instance="node3", type="logging"} level="info" Updating images
+2024-02-14T21:31:20Z {app="lxd", instance="node3", type="logging"} level="info" Done pruning resolved warnings
+2024-02-14T21:31:20Z {app="lxd", instance="node3", type="logging"} level="info" Done expiring log files
+2024-02-14T21:31:20Z {app="lxd", instance="node3", type="logging"} level="info" Done updating images
+...
+```
+
+## Adding labels
```
Comment (ru-fu) [inline doc/howto/logs_loki.md:51]:
```suggestion
## Add labels
```

File: doc/metrics.md
Comment (ru-fu) [inline doc/metrics.md:1]: Can you add a link to the Grafana how-to in the intro?

```
<!-- Include start metrics intro -->
LXD collects metrics for all running instances as well as some internal metrics.
These metrics cover the CPU, memory, network, disk and process usage.
They are meant to be consumed by Prometheus, and you can use Grafana to display the metrics as graphs.
See {ref}`provided-metrics` for lists of available metrics and {ref}`grafana` for instructions on how to display the metrics in Grafana.
<!-- Include end metrics intro -->
```
(this is from line 12-16, only the last part of the last sentence is new)

File: doc/howto/logs_loki.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+relatedlinks: https://grafana.com/grafana/dashboards/19131-lxd/
+---
+
+(logs_loki)=
+# How to send logs to Loki
+
+<!-- Include start logs_loki intro -->
+LXD publishes information about its activity in the form of events. The `lxc monitor` command allows you to view this information in your shell. There are two categories of LXD events: logs and life cycle. The `lxc monitor --type=logging --pretty` command will filter and display log type events like activity of the raft cluster for instance while `lxc monitor --type=logging --pretty` will only display life cycle events like instances starting or stopping.
+
+In a production environment, one might want to persist these events in a dedicated system. [Loki](https://grafana.com/oss/loki/) is one such system and LXD provides a configuration option to forward its event stream to Loki.
```
[the remark this answers is not in the document]
Comment (ru-fu) [reply doc/howto/logs_loki.md:11]: Add a link to the Grafana how-to here:

```
You can then display the log events in Grafana. See {ref}`grafana` for instructions.
```

File: doc/howto/grafana.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+- [Install Grafana](https://grafana.com/docs/grafana/latest/setup-grafana/installation/)
+- [Sign in to Grafana](https://grafana.com/docs/grafana/latest/setup-grafana/sign-in-to-grafana/)
+
+Complete the following steps to import the [LXD dashboard](https://grafana.com/grafana/dashboards/19131-lxd/):
+
+1. Configure Prometheus as a data source:
+
+   1. From the Basic (quick setup) panel, chose {guilabel}`Data Sources`.
+
+      ![Chose data source in Grafana](/images/grafana_welcome.png)
```
Comment (ru-fu) [inline doc/howto/grafana.md:26]:
```suggestion
   1. From the Basic (quick setup) panel, choose {guilabel}`Data Sources`.

      ![Choose data source in Grafana](/images/grafana_welcome.png)
```

**Review by ru-fu — COMMENTED** [review]
Thanks, this looks great!
Some comments for consistency.

**Comment by tomponline** [comment]
@escabo as @ru-fu has provided some feedback, I thought it would be prudent to let you know that because we dont use squash merges in LXD, the full commit list of your PR will land in the main branch of LXD. As such we don't want to see additional commits to show the fixes @ru-fu has suggested, but instead the changes should be squashed or fixed up into the existing commit(s).

As you've only got a single commit here, then `git commit --amend` could be useful.

**Review by tomponline — DISMISSED** [review]

**Review by tomponline — APPROVED** [review]

**Comment by tomponline** [comment]
I'll just wait for @ru-fu approval on the content

**Review by ru-fu — APPROVED** [review]
Thanks! :1st_place_medal:

**Pull request merged**